lightningd/chaintopology.c

```diff
@@ -100,11 +100,12 @@ static void filter_block_txs(struct chain_topology *topo, struct block *b)
 			const u8 *oscript = bitcoin_tx_output_get_script(tmpctx, tx, k);
 				if (txfilter_scriptpubkey_matches(topo->bitcoind->ld->owned_txfilter, oscript)) {
 					struct amount_sat amount;
-					struct bitcoin_outpoint outpoint;
-					outpoint.txid = txid;
-					outpoint.n = k;
+					struct bitcoin_outpoint *outpoint = tal(tmpctx, struct bitcoin_outpoint);
+					outpoint->txid = txid;
+					outpoint->n = k;
 					fprintf(stderr, "CAG txfilter_scriptpubkey_matches found for script %s\n", tal_hex(tmpctx, oscript));
 					bitcoin_tx_output_get_amount_sat(tx, k, &amount);
+					// fprintf(stderr, "CAG checking outpoint: %s\n", type_to_string(tmpctx, struct bitcoin_outpoint, outpoint));
 					invoice_check_onchain_payment(topo->ld, oscript, amount, outpoint);
```

Comment:
This change seems wrong...  Did you just want to hand &outpoint?

Reply:
IIRC the outpoint was getting dropped somewhere between here and when the notification is sent, so I needed to allocate it. Does handing `&outpoint` accomplish this as well?